cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml

```diff
@@ -169,144 +189,167 @@ data:
-      pos_file /var/log/es-kube-controller-manager.log.pos
+      pos_file /var/log/kube-controller-manager.log.pos
       tag kube-controller-manager
     </source>
 
     # Example:
     # W0204 06:49:18.239674       7 reflector.go:245] pkg/scheduler/factory/factory.go:193: watch of *api.Service ended with: 401: The event in requested index is outdated and cleared (the requested history has been cleared [2578313/2577886]) [2579312]
     <source>
-      type tail
+      @id kube-scheduler.log
+      @type tail
       format multiline
       multiline_flush_interval 5s
       format_firstline /^\w\d{4}/
       format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
       time_format %m%d %H:%M:%S.%N
       path /var/log/kube-scheduler.log
-      pos_file /var/log/es-kube-scheduler.log.pos
+      pos_file /var/log/kube-scheduler.log.pos
       tag kube-scheduler
     </source>
 
     # Example:
     # I1104 10:36:20.242766       5 rescheduler.go:73] Running Rescheduler
     <source>
-      type tail
+      @id rescheduler.log
+      @type tail
       format multiline
       multiline_flush_interval 5s
       format_firstline /^\w\d{4}/
       format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
       time_format %m%d %H:%M:%S.%N
       path /var/log/rescheduler.log
-      pos_file /var/log/es-rescheduler.log.pos
+      pos_file /var/log/rescheduler.log.pos
       tag rescheduler
     </source>
 
     # Example:
     # I0603 15:31:05.793605       6 cluster_manager.go:230] Reading config from path /etc/gce.conf
     <source>
-      type tail
+      @id glbc.log
+      @type tail
       format multiline
       multiline_flush_interval 5s
       format_firstline /^\w\d{4}/
       format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
       time_format %m%d %H:%M:%S.%N
       path /var/log/glbc.log
-      pos_file /var/log/es-glbc.log.pos
+      pos_file /var/log/glbc.log.pos
       tag glbc
     </source>
 
     # Example:
     # I0603 15:31:05.793605       6 cluster_manager.go:230] Reading config from path /etc/gce.conf
     <source>
-      type tail
+      @id cluster-autoscaler.log
+      @type tail
       format multiline
       multiline_flush_interval 5s
       format_firstline /^\w\d{4}/
       format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
       time_format %m%d %H:%M:%S.%N
       path /var/log/cluster-autoscaler.log
-      pos_file /var/log/es-cluster-autoscaler.log.pos
+      pos_file /var/log/cluster-autoscaler.log.pos
       tag cluster-autoscaler
     </source>
 
     # Logs from systemd-journal for interesting services.
     <source>
-      type systemd
+      @id journald-docker
+      @type systemd
       filters [{ "_SYSTEMD_UNIT": "docker.service" }]
-      pos_file /var/log/gcp-journald-docker.pos
+      #pos_file /var/log/journald-docker.pos
```

Comment:
do we not need the position file here?  What happens if the fluentd container gets restarted

Reply:
Sorry for barging in 😄 but since you have `read_from_head true`, I don't think it's wise to remove the `pos_file`, this combination will result in fluentd reading all the events in the journal when it is restarted as there's no reference where it last stopped, this could lead to huge bursts of re-ingested logs.

Reply:
pos_file is deprecated in the new plugin version. see: https://github.com/reevoo/fluent-plugin-systemd/blob/1e14ace403891b9aa745dc8a2a920103e5fda21e/README.md

> pos_file
> 
> This parameter is deprecated and will be removed in favour of storage in v1.0.
> 
> Path to pos file, stores the journald cursor. File is created if does not exist.
> 
> storage
> 
> Configuration for a storage plugin used to store the journald cursor.
> 
> Upgrading from pos_file
> 
> If pos_file is specified in addition to a storage plugin with persistent set to true, the cursor will be copied from the pos_file on startup, and the old pos_file removed.
> 
> pos_file
> 
> This parameter is deprecated and will be removed in favour of storage in v1.0.
> 
> Path to pos file, stores the journald cursor. File is created if does not exist.
> 
> storage
> 
> Configuration for a storage plugin used to store the journald cursor.
> 
> Upgrading from pos_file
> 
> If pos_file is specified in addition to a storage plugin with persistent set to true, the cursor will be copied from the pos_file on startup, and the old pos_file removed.

Reply:
@monotek you're absolutely right, but should you consider the "upgrade" scenario? keeping the exisiting `pos_file` configuration will ensure the cursor is copied to the new storage.

Also, shouldn't there be a `path` in the `<storage>` section?

Reply:
@dannyk81 
Not necessarily. I've used the root_dir option. See: https://docs.fluentd.org/v1.0/articles/storage_local

This is the reason for the new system.conf file: https://github.com/monotek/kubernetes/blob/0aff9bcf0402179c2dd5d11da31fccf3b228b5a1/cluster/addons/fluentd-elasticsearch/fluentd-es-configmap.yaml#L9

Beside that you'll get deprecation warnings when you still use the pos files. See:
https://github.com/reevoo/fluent-plugin-systemd/blob/dbd7359c6a8ce7e3b7ad9cfa447c35a5ec2fde04/lib/fluent/plugin/in_systemd.rb#L18

So i thought its better to have clean log when fluentd is started.